src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchCommand.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Configuration;
+using NuGet.Credentials;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal class PackageSearchCommand
+    {
+        internal delegate Task<int> SetupSettingsAndRunSearchAsyncDelegate(PackageSearchArgs packageSearchArgs);
```

Comment:
Why do we need this delegate? 
If I'm not wrong, SetupSettingsAndRunSearchAsyncDelegate is used everywhere. 

Why not just call the method?